platform/machine/do/cluster.go

```diff
@@ -87,8 +90,12 @@ func (dc *cluster) NewMachine(userdata *conf.UserData) (platform.Machine, error)
 		return nil, err
 	}
 
-	droplet, err := dc.api.CreateDroplet(context.TODO(), dc.vmname(), dc.sshKeyID, conf.String())
-	if err != nil {
+	var droplet *godo.Droplet
+	// DO frequently gives us 422 errors saying "Please try again"
+	if err := util.Retry(6, 10*time.Second, func() error {
+		droplet, err = dc.api.CreateDroplet(context.TODO(), dc.vmname(), dc.sshKeyID, conf.String())
+		return err
+	}); err != nil {
```

Comment:
We should only retry if it's the 422 error.

Reply:
I don't know. What if they return an error thats not 422 but still says retry? I don't see a ton of harm in retrying even if it'll just fail again. This would also retry in the case of 500 series errors, which imo we should do.

What about checking if its a `godo.ErrorResponse` (should cover any error returned from the api but not anything from the godo library itself)?

I'll also add logging, so we'll have a chance of seeing any errors.

Reply:
At a minimum we definitely need to log the errors we're getting before doing a retry. When discussing retries in the past with @bgilbert our thought was to only add a specific set of errors in the case of platform/api level retries that we know is a platform issue (availability, known flakes with their deployment) so that we don't accidentally suppress real failures.

Reply:
In the case of "not out fault" errors I think the correct action is to retry a couple times, then skip the test if it still fails. Those two are unrelated AFAICT. For the former (what is being addressed here) I think I'll add a `func shouldRetry(err error) bool` and maybe a `util.RetryConditional()` function.

For the latter (would be a seperate PR), it might be good to add a `IsCloudError(err error) bool` (not married to that name)  to the cluster interface that the harness can use to determine if the test should be skipped?

Reply:
I agree, as long as we scope down the errors that trigger `shouldRetry` to be true and add logging denoting the error and that a retry is occuring.

@bgilbert any thoughts on whether 500's are acceptable for retries? My initial thought is that they probably are okay to be retried initially and we can potentially scope it back further if we find out that legitimate errors are being suppressed.

Reply:
Absent platform documentation listing retryable errors, I'd generally prefer retrying only on errors we've specifically encountered and evaluated in the past, but I'm not tied to it.

The problem with something like `IsCloudError` is that we stringify and wrap errors all over the place.  It'd be easy to accidentally smash an error type during a refactor.